lib/packwerk/constant_discovery.rb

```diff
@@ -15,32 +15,50 @@ module Packwerk
   #   have no way of inferring the file it is defined in. You could argue though that inheritance means that another
   #   constant with the same name exists in the inheriting class, and this view is sufficient for all our use cases.
   class ConstantDiscovery
-    ConstantContext = Struct.new(:name, :location, :package, :public?)
+    extend T::Sig
 
-    # @param constant_resolver [ConstantResolver]
-    # @param packages [Packwerk::PackageSet]
+    class ConstantContext < T::Struct
+      extend T::Sig
+      const :name, String
+      const :location, String
+      const :package, Package
+      const :public, T::Boolean
+
+      sig { returns(T::Boolean) }
+      def public?
+        public
+      end
+    end
+
+    sig do
+      params(constant_resolver: ConstantResolver, packages: Packwerk::PackageSet).void
+    end
     def initialize(constant_resolver:, packages:)
       @packages = packages
       @resolver = constant_resolver
     end
 
-    # Get the package that owns a given file path.
-    #
-    # @param path [String] the file path
-    #
-    # @return [Packwerk::Package] the package that contains the given file,
-    #   or nil if the path is not owned by any component
+    # Get the package that owns a given file path, or the root package if the path is not owned by any component
+    sig do
+      params(
+        path: String,
+      ).returns(Packwerk::Package)
+    end
     def package_from_path(path)
       @packages.package_from_path(path)
     end
 
     # Analyze a constant via its name.
     # If the constant is unresolved, we need the current namespace path to correctly infer its full name
-    #
-    # @param const_name [String] The unresolved constant's name.
-    # @param current_namespace_path [Array<String>] (optional) The namespace of the context in which the constant is
-    #   used, e.g. ["Apps", "Models"] for `Apps::Models`. Defaults to [] which means top level.
-    # @return [Packwerk::ConstantDiscovery::ConstantContext]
+    sig do
+      params(
+        # The unresolved constant's name.
```

Comment:
Will this comment be presented in the documentation? If not we should not remove the documentation in favor of a comment in the type.

Reply:
I'm not totally following this one -- are you referring to the end-user facing documentation (README and USAGE)?

When you say a comment in the type -- I have a comment over the type right now, let me know what you mean by that or if what I have now works.